src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
+
+        [Fact]
+        public static void GenerateGuidForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateGuidForType(typeof(System.Int32)));        
```

Comment:
Can it just return Type.Guid then? I'm not aware of interesting uses, in fact I'd never really considered the API until seeing it in this PR. I'm simply highlighting that if we can easily implement an API we're adding back, that's much preferable to throwing.